i3_agenda/API.py

```diff
@@ -0,0 +1,131 @@
+
+import os
+import re
+from typing import List
+import datetime
+import pickle
+
+from google.auth.transport.requests import Request
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build, Resource
+
+from event import Event
+from event_utils import get_event_time, is_allday
+from config import CONF_DIR, URL_REGEX
+
+SCOPES = ["https://www.googleapis.com/auth/calendar.readonly"]
+TMP_TOKEN = f"{CONF_DIR}/i3agenda_google_token.pickle"
+
+
+def connect(credspath: str) -> Resource:
+    creds = None
+    # The file token.pickle stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if not os.path.exists(CONF_DIR):
+        os.mkdir(CONF_DIR)
+    if os.path.exists(TMP_TOKEN):
+        with open(TMP_TOKEN, "rb") as token:
+            creds = pickle.load(token)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if not os.path.exists(credspath):
+            print(
+                '''You need to download your credentials json file from the 
+                   Google API Console and pass its path to this script'''
+            )
+            exit(1)
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(credspath, SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open(TMP_TOKEN, "wb") as token:
+            pickle.dump(creds, token)
+
+    service = build("calendar", "v3", credentials=creds)
+    return service
+
+
+def getEvents(
```

Comment:
this is a very long function, could probably be separated into 3-4 additional functions

Reply:
I separated It too into more functions. A feedback will be welcome on this too.